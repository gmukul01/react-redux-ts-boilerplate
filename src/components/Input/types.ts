import { HTMLProps, WithThemeProp } from '../../utils/types';

export interface Props extends HTMLProps<HTMLInputElement>, WithThemeProp {}
