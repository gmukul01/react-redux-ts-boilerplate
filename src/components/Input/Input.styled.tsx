import { defaultTheme } from '@theme';
import { styled } from '@utils/styled';
import { Props } from './types';

export const InputStyled = styled('input').attrs(({ theme: { input, text, font } }) => ({ input, text, font }))<Props>`
    padding: 10px;
    font-size: ${props => props.text.defaults.textSize};
    font-weight: ${props => props.text.defaults.textWeight};
    color: ${props => props.input.color};
    background-color: ${props => props.input.bgColor};
    background-clip: padding-box;
    box-sizing: border-box;
    border: 1px solid ${props => props.input.borderColor};
    border-radius: 3px;

    &:focus {
        border-color: ${props => props.input.focus.borderColor};
        outline: 0;
    }

    &::placeholder {
        color: ${props => props.input.placeHolderColor};
    }
`;

InputStyled.defaultProps = {
    theme: defaultTheme
};
