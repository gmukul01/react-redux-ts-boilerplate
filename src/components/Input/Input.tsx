import React from 'react';
import { WithStyle } from '../../utils/types';
import { InputStyled } from './Input.styled';
import { Props } from './types';

const Input: React.FunctionComponent<Props> & WithStyle = React.forwardRef((props, ref) => {
    return <InputStyled ref={ref} {...props} />;
});

Input.displayName = 'Input';

export default Input;
