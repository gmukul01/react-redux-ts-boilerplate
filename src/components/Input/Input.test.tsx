import { shallow } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import { create } from 'react-test-renderer';
import Input from './Input';

describe('Input component', () => {
    it('should render default variant correctly', () => {
        const input = create(<Input type="email" />);
        expect(input).toMatchSnapshot();
    });
});
