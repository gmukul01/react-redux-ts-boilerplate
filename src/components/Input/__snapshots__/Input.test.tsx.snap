// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render default variant correctly 1`] = `
.c0 {
  padding: 10px;
  font-size: M2;
  font-weight: Normal;
  color: #2a2a2e;
  background-color: #ffffff;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #dddfe2;
  border-radius: 3px;
}

.c0:focus {
  border-color: #5392f9;
  outline: 0;
}

.c0::-webkit-input-placeholder {
  color: #d7d7db;
}

.c0::-moz-placeholder {
  color: #d7d7db;
}

.c0:-ms-input-placeholder {
  color: #d7d7db;
}

.c0::placeholder {
  color: #d7d7db;
}

<input
  className="c0"
  type="email"
/>
`;
